multicluster/controllers/multicluster/member/serviceexport_controller.go

```diff
@@ -265,37 +278,46 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			apiequality.Semantic.DeepEqual(svc.Spec.ClusterIPs, installedSvc.clusterIPs) {
 			klog.InfoS("Service has been converted into ResourceExport and no change, skip it", "service",
 				req.String(), "resourceexport", svcExportNSName)
-			svcNoChange = true
+			skipUpdateSvcResourceExport = true
 		}
 	}
 
-	if r.endpointIPType == "ClusterIP" {
-		svcIPAsSubset := common.GetServiceEndpointSubset(svc)
+	var newSubsets []corev1.EndpointSubset
+	if r.endpointIPType == common.EndpointIPTypeClusterIP {
+		svcIPAsSubset := getClusterIPEndpointSubset(svc)
 		if len(svcIPAsSubset.Addresses) == 0 {
 			err = r.updateSvcExportStatus(ctx, req, serviceNoClusterIP)
 			if err != nil {
 				return ctrl.Result{}, err
 			}
 			return ctrl.Result{}, nil
 		} else {
-			eps.Subsets = []corev1.EndpointSubset{svcIPAsSubset}
+			newSubsets = []corev1.EndpointSubset{svcIPAsSubset}
 		}
 	} else {
-		newSubsets := common.FilterEndpointSubsets(eps.Subsets)
-		if epsInstalled {
-			installedEp := epsObj.(*epInfo)
-			if apiequality.Semantic.DeepEqual(newSubsets, installedEp.subsets) {
-				klog.InfoS("Endpoints has been converted into ResourceExport and no change, skip it", "endpoints",
-					req.String(), "resourceexport", epExportNSName)
-				epNoChange = true
-			}
+		if r.endpointSliceEnabled {
+			newSubsets = eps.Subsets
+		} else {
+			newSubsets = filterEndpointSubsets(eps.Subsets)
 		}
-		if !epNoChange {
-			eps.Subsets = newSubsets
+	}
+
+	if epsInstalled {
+		installedEp := epsObj.(*epInfo)
+		if apiequality.Semantic.DeepEqual(newSubsets, installedEp.subsets) {
+			klog.InfoS("Service's Endpoints (PodIP or ClusterIP) has been converted into ResourceExport and no change, skip it", "Service",
+				req.String(), "resourceexport", epExportNSName)
+			// When the EndpointIPType is EndpointIPTypeClusterIP, skipUpdateEPResourceExport should be false only
+			// when there is a ClusterIP/Port change.
+			skipUpdateEPResourceExport = true
 		}
 	}
 
-	if svcNoChange && epNoChange {
+	if !skipUpdateEPResourceExport {
+		eps.Subsets = newSubsets
```

Comment:
This can be merged into the block at line 340, before `r.serviceHandler()` is called.

Reply:
Done, thanks.